src/metabase/driver/postgres.clj

```diff
@@ -165,6 +166,66 @@
   (binding [*enum-types* (enum-types driver database)]
     (sql-jdbc.sync/describe-table driver database table)))
 
+(def ^:const json-sample-limit
+  "Number of rows to sample for describe-table-json."
+  10000)
+
+(defn- flatten-row [row]
+  (letfn [(flatten-row [row path]
+            (lazy-seq
+              (when-let [[[k v] & xs] (seq row)]
+                (cond (and (map? v) (not-empty v))
+                      (into (flatten-row v (conj path k))
+                            (flatten-row xs path))
+                      :else
+                      (cons [(conj path k) v]
+                            (flatten-row xs path))))))]
+    (into {} (flatten-row row []))))
+
+(defn- row->types [row]
+  (into {} (for [[field-name field-val] row]
+             [field-name (let [flattened-row (flatten-row field-val)]
+                           (into {} (map (fn [[k v]] [k (type v)]) flattened-row)))])))
+
+(defn- describe-json-xform [member]
+  ((comp (map #(for [[k v] %] [k (json/parse-string v)]))
+         (map #(into {} %))
+         (map row->types)) member))
+
+(defn- describe-json-rf
+  ([] nil)
+  ([fst] fst)
+  ([fst snd]
+   (into {}
+         (for [json-column (keys snd)]
+           (if (or (nil? fst) (= (hash (fst json-column)) (hash (snd json-column))))
+             [json-column (snd json-column)]
+             [json-column nil])))))
+
+(defn- describe-table-json*
+  [driver spec table]
+  (with-open [conn (jdbc/get-connection spec)]
+    (let [map-inner        (fn [f xs] (map #(into {}
+                                                  (for [[k v] %]
+                                                    [k (f v)])) xs))
+          table-fields     (sql-jdbc.sync/describe-table-fields driver conn table)
+          json-fields      (filter #(= (:semantic-type %) :type/SerializedJSON) table-fields)
+          json-field-names (mapv (comp keyword :name) json-fields)
+          query-vec        (hsql/format {:select json-field-names
+                                         :from   [(keyword (:name table))]
+                                         :limit  json-sample-limit})
+          query            (jdbc/reducible-query spec query-vec)]
```

Comment:
```suggestion
          sql-args         (hsql/format {:select json-field-names
                                         :from   [(keyword (:name table))]
                                         :limit  json-sample-limit})
          query            (jdbc/reducible-query spec sql-args)]
```

I know what you mean by `query-vec` but `sql-args` is pretty much a standard idiomatic name for that in the HoneySQL/Clojure JDBC world and what we use elsewhere in the codebase.

Reply:
Also, you probably need to be passing `:quoting :ansi` to `hsql/format`, otherwise it's not going to quote identifiers. and if your identifier contains hyphens or dots or spaces or other uppercase characters or stuff like that this will break

Reply:
Looks like it does break on needs-quote stuff, thanks for catching